app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Magento\Catalog\Block\Adminhtml\Product\Edit\Button;
+
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable;
```

Comment:
Configurable product cannot be referred in Catalog module.

Also, all other composite product types which may contain simple product must be taken into consideration: grouped, bundle.